src/bootstrap/check.rs

```diff
@@ -120,7 +120,9 @@ impl Step for Std {
             target,
             cargo_subcommand(builder.kind),
         );
-        cargo.arg("--all-targets");
+        if builder.kind != Kind::Clippy {
+            cargo.arg("--all-targets");
+        }
```

Comment:
Hmm, this makes it so it's no longer possible to pass --all-targets to clippy - is that intentional? I agree it doesn't seem super useful, but it seems weird to remove the option altogether ... although I guess you can still add it back with `CARGOFLAGS=--all-targets`.

Reply:
Also, for this step specifically it should just be an early return above, because we've already run on the standard library; this invocation is *only* for the other targets. The other steps are correct though.

Reply:
How did that work previously? When I check ` ./x.py clippy --all-targets` pre-branch,  I just get a 
````
Unrecognized option: 'all-targets'

Usage: x.py <subcommand> [options] [<paths>...]
````
error...

Reply:
I think at least a comment as to why this isn't being specified is a good idea. We can consider the defaults (or ability to provide it) later.

Reply:
> How did that work previously?

See the changes in https://github.com/rust-lang/rust/pull/88011/files.

Reply:
Sure, added some comments.